### PR TITLE
Fixes Promscale on TimescaleDB-OSS.

### PR DESCRIPTION
**Pull request by Harkishen-Singh**
Signed-off-by: Harkishen Singh <harkishensingh@hotmail.com>

Fixes: #583 

**Note:** I tried to apply conditional checks where I felt would be necessary. There are high chances that I have missed them out at important places, please keep this in mind while reviewing.

TODO:

- [x] E2E for TimescaleDB-OSS version

### CODE REVIEW

**Review by antekresic — CHANGES_REQUESTED** [review]
Missing check in `config_maintenance_jobs`, uses user-defined actions which are not in OSS:
https://docs.timescale.com/api/latest/actions/

**Review by cevian — CHANGES_REQUESTED** [review]

**Review by cevian — APPROVED** [review]
Looks good. I do however want to add a big WARNING when promscale first starts up if we are using OSS to warn people: 
```
WARNING: Using the Apache2 version of TimescaleDB. This version does not include compression and thus performance and disk usage will be significantly negatively effected.
```

File: pkg/migrations/sql/idempotent/base.sql [hunk 1]
```diff
@@ -27,13 +27,35 @@ LANGUAGE SQL STABLE PARALLEL SAFE;
 GRANT EXECUTE ON FUNCTION SCHEMA_CATALOG.get_default_retention_period() TO prom_reader;
 
 CREATE OR REPLACE FUNCTION SCHEMA_CATALOG.is_timescaledb_installed()
-    RETURNS BOOLEAN
+RETURNS BOOLEAN
 AS $func$
     SELECT count(*) > 0 FROM pg_extension WHERE extname='timescaledb';
-$func$
+    $func$
```
Comment (antekresic) [inline pkg/migrations/sql/idempotent/base.sql:33]: Minor nit: why add whitespace here? It even makes things harder to read.

File: pkg/runner/client.go [hunk 2]
```diff
@@ -177,6 +186,26 @@ func CreateClient(cfg *Config, promMetrics *api.Metrics) (*pgclient.Client, erro
...
 }
 
+func isTimescaleDBOSS(conn *pgx.Conn) (bool, error) {
+	var (
+		isTimescaleDB bool
+		isLicenseOSS  bool
+	)
+	err := conn.QueryRow(context.Background(), "SELECT "+schema.Catalog+".is_timescaledb_installed()").Scan(&isTimescaleDB)
+	if err != nil {
+		return false, fmt.Errorf("fetching if TimescaleDB is installed: %w", err)
```
Comment (antekresic) [inline pkg/runner/client.go:196]:
```suggestion
		return false, fmt.Errorf("error fetching if TimescaleDB is installed: %w", err)
```

File: pkg/runner/client.go [hunk 3]
```diff
@@ -177,6 +186,26 @@ func CreateClient(cfg *Config, promMetrics *api.Metrics) (*pgclient.Client, erro
...
+	}
+	if !isTimescaleDB {
+		// Return false so that we don't warn for OSS TimescaleDB.
+		return false, nil
+	}
+	err = conn.QueryRow(context.Background(), "SELECT "+schema.Catalog+".is_timescaledb_oss()").Scan(&isLicenseOSS)
+	if err != nil {
+		return false, fmt.Errorf("fetching TimescaleDB license: %w", err)
```
Comment (antekresic) [inline pkg/runner/client.go:204]:
```suggestion
		return false, fmt.Errorf("error fetching TimescaleDB license: %w", err)
```

**Review by antekresic — APPROVED** [review]
Some minor nits, should be good to go.

**Pull request merged**